src/main/java/com/aws/greengrass/tes/CredentialRequestHandler.java

```diff
@@ -314,10 +314,11 @@ private byte[] getCredentialsBypassCache() {
             tesCache.get(iotCredentialsPath).expiry = newExpiry;
             tesCache.get(iotCredentialsPath).credentials = response;
         } catch (AWSIotException | TLSAuthException e) {
-            // Http connection error should expire immediately
+            // Http connection error should be cached to avoid excessive retries
             String responseString = "Failed to get connection";
             response = responseString.getBytes(StandardCharsets.UTF_8);
-            newExpiry = Instant.now(clock);
+            // Use unknown error cache policy for SSL/TLS connection errors to prevent excessive retries
+            newExpiry = Instant.now(clock).plus(Duration.ofMinutes(UNKNOWN_ERROR_CACHE_IN_MIN));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

These configs are flagged under GS-Tech Static config campaign and is considered to be a violation. Please migrate these configs